lightning/src/chain/channelmonitor.rs

```diff
@@ -1973,21 +1965,25 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 		}
 
 		self.onchain_tx_handler.block_connected(txn_matched, claimable_outpoints, height, &*broadcaster, &*fee_estimator, &*logger);
+		self.last_block_hash = block_hash;
 
-		self.last_block_hash = block_hash.clone();
-		for &(ref txid, ref output_scripts) in watch_outputs.iter() {
-			self.outputs_to_watch.insert(txid.clone(), output_scripts.iter().map(|o| o.script_pubkey.clone()).collect());
-		}
-
+		watch_outputs.retain(|&(ref txid, ref txouts)| {
```

Comment:
A comment similar to the commit message explaining this block help readability.

Reply:
Good point! Done.